py4DSTEM/process/diskdetection/diskdetection.py

```diff
@@ -566,5 +595,146 @@ def threshold_Braggpeaks(pointlistarray, minRelativeIntensity, relativeToPeak, m
+                    deletemask = pointlist.data['intensity'] / mean_intensity < \
+                                                                               minIntensity
+                    pointlist.remove_points(deletemask)
+                
+                if metric == 'maximum':
+                    deletemask = pointlist.data['intensity'] / max_intensity < \
+                                                                               minIntensity
+                    pointlist.remove_points(deletemask)
+                    
+                if metric == 'median':
+                    deletemask = pointlist.data['intensity'] / median_intensity < \
+                                                                               minIntensity
+                    pointlist.remove_points(deletemask)
+                    
+                if metric == 'manual':
+                    deletemask = pointlist.data['intensity'] < minIntensity
+                    
+                    pointlist.remove_points(deletemask)
+            
+            # Remove peaks that are too close together
+            if maxNumPeaks is not False:
+                r2 = minPeakSpacing**2
+                deletemask = np.zeros(pointlist.length, dtype=bool)
+                for i in range(pointlist.length):
+                    if deletemask[i] == False:
+                        tooClose = ( (pointlist.data['qx']-pointlist.data['qx'][i])**2 + \
+                                     (pointlist.data['qy']-pointlist.data['qy'][i])**2 ) < r2
+                        tooClose[:i+1] = False
+                        deletemask[tooClose] = True
+                pointlist.remove_points(deletemask)
+
+            # Keep only up to maxNumPeaks
+            if maxNumPeaks is not False:
+                if maxNumPeaks < pointlist.length:
+                    deletemask = np.zeros(pointlist.length, dtype=bool)
+                    deletemask[maxNumPeaks:] = True
+                    pointlist.remove_points(deletemask)
+
+    return pointlistarray
+
 
+def get_pointlistarray_hist(pointlistarray, return_hist = False, bins = 200):
+    
+    """
+    Concatecates the Bragg peak intensities from a PointListArray of Bragg peak positions into one array and returns the counts and bins. 
+    This output can be used for understanding the distribution of intensities in your dataset for universal thresholding.
+    
+    Accepts:
+        pointlistarray      (PointListArray)
+        return_hist         (bool) if True, returns histogram of intensities in the entire pointlist
+        bins                (int) number of bins that the intensity values will be sorted into for histogram 
+    
+    Returns:
+        peak_intensities    (ndarray) all detected peaks
+        
+        If return_hist==False (default), the figure is plotted and only peak intensities are returned.
+        If return_hist==True, the counts and bins are returned, and can be manually plotted.
```

Comment:
Suggest
```
counts        (ndarray) returned iff return_hist==True. The histogram bin counts
bin_edges  (ndarray) returned iff return_hist==True. The histogram bin edges
```

I would also strongly suggest we rename `bin_values` to either `bin_edges` or `bin_centers`, depending on which we're returning.  Right now it seems we're returning the edges (which is good, this is what numpy users will be used to / expect, even if this convention is sometimes frustrating / dumb.  Please see my comment in the visualize function!)